### PR TITLE
`[Navigation.Item]` Fix navigation sub item selected indicator when no nav icon

### LINKED ISSUE #8342: Navigation item bug with no icon
**[Alex Montague](https://github.com/alexanderMontague)** [2023-02-03 07:54PM UTC]<br/>
Hey Polaris team :wave:  I have a question about how nav items and specifically the sub nav items should behave. For context, we have a takeover nav in Partners that shows when you are in the context of an app, where this behaviour is happening.

Basically, when you have a nav item with `subNavigationItems` , the highlighted tab that indicates if the sub nav item is active or not does not get pushed out to the full left margin if the parent nav item does not have an icon.

<details>
<summary>:thread: Slack Thread</summary>
<table>
<thead>
<tr><th>

[Slack Thread](https://shopify.slack.com/archives/C4Y8N30KD/p1675454091990829?thread_ts=1675454091.990829&cid=C4Y8N30KD)

</th></tr>
</thead>
<tbody>
<tr>
<td>

 **[Alex Montague](https://github.com/alexanderMontague)** [2023-02-03 07:54PM UTC]<br/>
Hey Polaris team :wave:  I have a question about how nav items and specifically the sub nav items should behave. For context, we have a takeover nav in Partners that shows when you are in the context of an app, where this behaviour is happening.

Basically, when you have a nav item with `subNavigationItems` , the highlighted tab that indicates if the sub nav item is active or not does not get pushed out to the full left margin if the parent nav item does not have an icon.

</td>
</tr>
<tr>
<td>

 **[Alex Montague](https://github.com/alexanderMontague)** [2023-02-03 07:55PM UTC]<br/>
I sent this message by mistake haha, but let me attach some images

</td>
</tr>
<tr>
<td>

 **[Alex Montague](https://github.com/alexanderMontague)** [2023-02-03 07:55PM UTC]<br/>
Without a parent nav item icon, this is what it looks like (you can see the green indicator does not hug the left margin) [image.png](https://files.slack.com/files-pri/T017D49VC3F-F04MU8DA5GW/image.png) 

</td>
</tr>
<tr>
<td>

 **[Alex Montague](https://github.com/alexanderMontague)** [2023-02-03 07:56PM UTC]<br/>
But with a top level nav item icon, it gets pushed out [image.png](https://files.slack.com/files-pri/T017D49VC3F-F04N0S3F7FU/image.png) 

</td>
</tr>
<tr>
<td>

 **[Alex Montague](https://github.com/alexanderMontague)** [2023-02-03 07:57PM UTC]<br/>
I've jigged some custom css to achieve our current designs (this is part of a bigger webification project in partners) to accomplish what I think it should look like. And [here is my PR](https://github.com/Shopify/partners-web-platform/pull/390) for even more context [image.png](https://files.slack.com/files-pri/T017D49VC3F-F04MU8LFD46/image.png) 

</td>
</tr>
<tr>
<td>

 **[Alex Montague](https://github.com/alexanderMontague)** [2023-02-03 07:57PM UTC]<br/>
so TLDR, is this expected and if its not I'd be happy to make a PR!

</td>
</tr>
<tr>
<td>

 **[Raman Lally](https://github.com/Rmnlly)** [2023-02-03 08:25PM UTC]<br/>
Oh this is interesting, I was checking the polaris storybook and it seems there aren’t really any examples of nav without icons so I can see how this might’ve slipped by. I have a feeling this is unintentional but of course I can’t speak too definitively on it!

</td>
</tr>
<tr>
<td>

 **[Raman Lally](https://github.com/Rmnlly)** [2023-02-03 08:36PM UTC]<br/>
Having worked on the nav items a lot, lemme know if you want to pair on this if you do end up making a PR for it though! :meowthumbsup:

</td>
</tr>
</tbody>
</table>
</details>

### PR DESCRIPTION
**Pull request by alexanderMontague**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/8342

Before:

Screenshot 2023-02-14 at 10 32 05 AM

Screenshot 2023-02-14 at 10 32 16 AM

After:

image

image

### WHAT is this pull request doing?

- While building a nav in Partners I stumbled across this weird behaviour, more details in the linked issue above
- TLDR: when a nav item does not have an icon and there are sub items, if one of those sub items are selected the selector indicator (green indicator on left of item) does not fully hug the leftmost margin
- This was due to legacy no icon styles that were introduced a while ago
- See this PR that added in these styles https://github.com/Shopify/polaris/pull/2874

### How to 🎩

- head to my spin http://240.26.37.19:6006/?path=/story/playground--nav-1
- toggle between `nav-1` and `nav-2` which are the navs with an icon and without
- they should now have the same spacing and the selector should fully hug the left margin

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [x] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Comment by kyledurand** [comment]
cc @henryyi just want to make sure this doesn't affect any of your nav work

**Review by sophschneider — APPROVED** [review]
Thanks for fixing this!

**Comment by henryyi** [comment]
Not sure if intended but the change causes the sub-nav indentation to be wider than it is currently:
in prod:
prod

in spin:
spin

**Comment by alexanderMontague** [comment]
@henryyi how did you come across this? Unless I'm missing something it seems to be the same:

**In Spin/storybook:**
image
 (40/4 padding)
image


<hr />

**In prod/web:**
image (40/4 padding)
image

**Comment by henryyi** [comment]
> @henryyi how did you come across this? Unless I'm missing something it seems to be the same:

This happens in a store with the global-nav. You can check this test store: https://henry-dev2.myshopify.com/admin
Also in this sandbox with latest polaris: https://codesandbox.io/s/divine-hill-2w33ru

**Comment by alexanderMontague** [comment]
@henryyi you are correct, this seems to only be happening for the plus side nav there. This brings into question (since I've never actually seen this nav view before), the highlight bug is actually in effect in this nav as you can see here:

image


image

The sub item spacing has not been consistent with the rest of Polaris Icon sub nav items. So this actually makes them consistent with that, but _should_ this be the case? I wouldn't mind some design input on this. 

Basically, should the sub nav items all have the same left indent regardless of if the parent has an icon or not, or should there be less padding like it currently is on that plus nav right now

**Comment by henryyi** [comment]
@dustinmalik do you mind having a look to determine what the preferred indentation is when we are not using icons (e.g when global-nav is visible).

**Comment by alexanderMontague** [comment]
For more context, I was able to boot this view up in spin with the change. This is what it looks like on this view:
image

image

It does look a little too justified without the icon, but this is the same indent as web

**Comment by alexanderMontague** [comment]
Any updates @henryyi @dustinmalik ?

**Comment by henryyi** [comment]
> Any updates @henryyi @dustinmalik ?

I'll wait for Dustin's input but my gut feeling is that we'll want to keep the lesser spacing since that is what merchants currently see despite the selection bug.

**Comment by dustinmalik** [comment]
Tried finding old Figma files for this but couldn't find anything. I would keep the sub nav items flush with the rest of the nav items like when there is an icon. 

image

@alexblaise was the designer on this. Thoughts?

**Comment by dustinmalik** [comment]
Discussed with @chrisblinstrub. Let's go with 24px padding. Thanks!

image

**Comment by alexanderMontague** [comment]
Thanks for the follow up everyone, appreciate it!! @henryyi ready for review again:

Storybook:
image
image

http://240.26.37.19:6006/?path=/story/playground--nav-1

Admin:
image

https://admin.web.web-nths.alex-montague.us.spin.dev/store/shop2/draft_orders?selectedView=all

Plus admin nav:
image

https://admin.web.plus-home-k10h.alex-montague.us.spin.dev/store/shop5/draft_orders

image

**Review by henryyi — APPROVED** [review]
Looks good!
